modules/qna/src/backend/storage.ts

```diff
@@ -33,271 +33,257 @@ const normalizeQuestions = (questions: string[]) =>
+          await fse.writeFile(path.resolve(tmpDir.name, mediaFile), await this.ghost.readFileAsBuffer('/', mediaFile))
+      )
+    )
 
-  async getAllContentElementIds(list?: QnaItem[]): Promise<string[]> {
-    const qnas = list || (await this.fetchQNAs())
-    const allAnswers = _.flatMapDeep(qnas, qna => Object.values(qna.data.answers))
-    return _.uniq(_.filter(allAnswers as string[], x => _.isString(x) && x.startsWith('#!')))
+    await tar.create({ cwd: tmpDir.name, file: zipName, portable: true, gzip: true }, [...medias, QNA_PATH_TO_ZIP])
+    const zipBuffer = await fse.readFile(zipName)
+    tmp.setGracefulCleanup()
+    return zipBuffer
   }
 
-  async getCountByTopic(): Promise<{ [context: string]: number }> {
-    const qnas = await this.fetchQNAs()
+  async importPerTopic(importArgs: ImportArgs) {
+    const { topicName, botId, zipFile } = importArgs
 
-    return _.countBy(qnas, x => x.data.contexts)
-  }
+    const tmpDir = tmp.dirSync({ unsafeCleanup: true })
+    await fse.writeFile(path.join(tmpDir.name, toZipFile(topicName)), zipFile)
+    await tar.extract({ cwd: tmpDir.name, file: path.join(tmpDir.name, toZipFile(topicName)) })
 
-  async getContentElementUsage(): Promise<any> {
-    const qnas = await this.fetchQNAs()
-
-    return _.reduce(
-      qnas,
-      (result, qna) => {
-        const answers = _.flatMap(Object.values(qna.data.answers))
-
-        _.filter(answers, x => x.startsWith('#!')).forEach(answer => {
-          const values = result[answer]
-          if (values) {
-            values.count++
-          } else {
-            result[answer] = { qna: qna.id, count: 1 }
-          }
-        })
-        return result
-      },
-      {}
+    let qnasBotAgnostic = JSON.parse(
+      await fse.readFile(path.join(tmpDir.name, FLOW_FOLDER, TEMP_INTENT_FILENAME), 'utf8')
     )
-  }
-
-  async count() {
-    const questions = await this.fetchQNAs()
-    return questions.length
-  }
+    qnasBotAgnostic = await Joi.array()
+      .items(QnASchema)
+      .validate(qnasBotAgnostic)
+
+    const qnaForBot = qnasBotAgnostic.map(i => {
+      const newContent = [...i.metadata.contentAnswers].map(c => addBotPrefix(c, botId))
+      const newMetadata = { ...i.metadata, contentAnswers: newContent }
+      return { ...i, contexts: [topicName], filename: path.join(topicName, INTENT_FILENAME), metadata: newMetadata }
+    })
 
-  // TODO remove batch deleter, it's done one by one anyway
-  async delete(qnaId) {
-    const ids = _.isArray(qnaId) ? qnaId : [qnaId]
-    if (!ids.length) {
-      return
-    }
+    await this.ghost.upsertFile(FLOW_FOLDER, path.join(topicName, INTENT_FILENAME), serialize(qnaForBot))
 
-    const deletePromise = async (id: string): Promise<void> => {
-      await this.deleteMatchingIntent(id)
-      return this.bp.ghost.forBot(this.botId).deleteFile(this.config.qnaDir, `${id}.json`)
+    for (const mediaFile of fse.readdirSync(path.join(tmpDir.name, MEDIA_FOLDER))) {
```

Comment:
`await fse.readdir`

Also, you don't check if the folder exists prior to reading it. If you don't have any media, then the folder will not be created in the archive and it will fail